### ISSUE TITLE
Non-IO show calls in show overload

### ISSUE THREAD

**Issue by TotalVerb** [tonyhffong/Lint.jl#163]
I make this mistake a lot:

```julia
Base.show(io::IO, x::MyType) = print(x.data)
```

Note that should have been `print(io, x.data)`. We could warn when an IO function called within a `show` overload does not seem to have the IO object as its first argument.
